symengine/pow.cpp

```diff
@@ -311,29 +323,52 @@ RCP<const Basic> Log::create(const RCP<const Basic> &a) const
 
 RCP<const Basic> log(const RCP<const Basic> &arg)
 {
-    if (eq(*arg, *zero)) {
-        throw NotImplementedError(
-            "log(0) is complex infinity. Yet to be implemented");
-    }
+    if (eq(*arg, *zero))
+        return ComplexInf;
     if (eq(*arg, *one))
         return zero;
     if (eq(*arg, *E))
         return one;
+    if (eq(*arg, *Inf))
```

Comment:
@ranjithkumar007, no you don't need to for this case. `get_eval` is need for functions like `sin`, `cos`, etc.